gslib/commands/du.py

```diff
@@ -231,14 +216,12 @@ def RunCommand(self):
         elif o == '-X':
           if a == '-':
             f = sys.stdin
+            f_close = False
           else:
-            f = open(a, 'r', encoding=UTF8)
-          position = f.tell()
-          try:
-            self.exclude_patterns = _GetPatternExclusion(f, position, UTF8)
-          except:
-            self.exclude_patterns = _GetPatternExclusion(f, position, LOCAL_ENC)
-          finally:
+            f = open(a, 'r') if six.PY2 else open(a, 'r', encoding=UTF8)
+            f_close = True
+          self.exclude_patterns = [six.ensure_text(line.strip()) for line in f]
```

Comment:
Q: The locale stuff has been removed in your changes -- `ensure_text` decodes with utf-8 unless otherwise specified; are there any cases where we want to use the system's preferred encoding instead of utf-8?

Reply:
As far as our testing goes, there doesn't appear to be any need for system-preferred encoding. I'm not sure if that would limit the types of files that could be read, and if it does, how likely one of those other types would be used.